docs/architecture/adr-042-state-sync.md

```diff
@@ -0,0 +1,119 @@
+# ADR 042: State Sync Design
+
+## Changelog
+
+2019-06-27: Init by EB
+
+## Context
+
+"State Sync" would allow new nodes to skip downloading all the blocks and
+just download a recent application state. Nodes would first perform a
+light-client sync to find a recent height for which they could sync the state.
+Then they would request verifiable pieces of that state concurrently from their peers.
+Once they have a complete state, they can use "Block Sync" to download
+and execute any more recent blocks.
+
+Note that Tendermint does not actually know anything about the application
+state; it is only available over ABCI. State Sync will thus require new ABCI
+methods and/or connections, and will need to support arbitrary Merkle trees.
+
```

Comment:
Suggested addition:

Tendermint is responsible for downloading and verifying chunks of `AppState` from peers. ABCI Application is responsible for taking `AppStateChunk` objects from TM and constructing a valid state tree whose root corresponds with the `AppHash` of syncing block.

Aside:

This is only true of `AppStateChunks` correct? Any `BlockChunk` or `StateChunk` that gets sent by peers can be handled entirely in TM

Reply:
Right